src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsControllerTest.java

```diff
@@ -206,6 +213,75 @@ void should_return_a_400_when_restricted_role_is_given() throws Exception {
+
+            Map<String, String> roleAttributes = new HashMap<>();
+            roleAttributes.put(RoleAttributeDefinition.JURISDICTION.value(), "IA");
+
+            // Role attribute is IA
+            List<Assignment> allTestRoles = new ArrayList<>();
+            roleNames.forEach(roleName -> asList(RoleType.ORGANISATION, RoleType.CASE)
+                .forEach(roleType -> {
+                    Assignment roleAssignment = mockServices.createBaseAssignment(
+                        UUID.randomUUID().toString(), "tribunal-caseworker",
+                        roleType,
+                        Classification.PUBLIC,
+                        roleAttributes
+                    );
+                    allTestRoles.add(roleAssignment);
+                }));
+
+            GetRoleAssignmentResponse accessControlResponse = new GetRoleAssignmentResponse(
+                allTestRoles
+            );
+            when(roleAssignmentServiceApi.getRolesForUser(
+                any(), any(), any()
+            )).thenReturn(accessControlResponse);
+
+            final String taskId = UUID.randomUUID().toString();
+
+            final var warningList = new WarningValues(asList(new Warning("TA01", "Description")));
+            CamundaTask task = new CamundaTask(taskId,
+                                               "name",
+                                               "assignee",
+                                               ZonedDateTime.now(),
+                                               ZonedDateTime.now(),
+                                               "description",
+                                               "owner",
+                                               "formkey",
+                                               "processInstanceId");
+
+            when(idamWebApi.token(any())).thenReturn(new Token(userToken, "scope"));
+            when(camundaServiceApi.getTask(any(), any()))
+                .thenReturn(task);
+
+            mockCamundaVariablesWithWarning();
+
+            mockMvc.perform(
+                get("/task/" + taskId)
+                    .header(
+                        "Authorization",
+                        authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+                    )
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+            ).andExpect(status().isOk())
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(jsonPath("$.task.id").value(taskId))
+                .andExpect(jsonPath("$.task.warnings").value(true))
+                .andExpect(jsonPath("$.task.warning_list.values").isArray())
+                .andExpect(jsonPath("$.task.warning_list.values",hasSize(1)))
+                .andExpect(jsonPath("$.task.warning_list.values[0].warningCode").value("TA01"))
+                .andExpect(jsonPath("$.task.warning_list.values[0].warningText").value("Description"));
+        }
+
```

Comment:
Can we also adda test to validate warnings are applied to tasks with multiple task categories?

Reply:
A ticked has been raised to deal with tasks with multiple categories